determined-ai/determined
# chore: add regex in `InlineEditor` [DET-7518]

## Description
[DET-7518]


## Screenshot


https://user-images.githubusercontent.com/109113616/182768603-4b33765a-742e-45dd-a44f-5c92fffb384c.mov



## Test Plan
- Check if InlineEditor works correctly
- Check if display name editor works correctly 
  - should shake when its empty or non alphanumeric is used, and previous value should be set
  - should save data when input is alphanumeric



## Commentary (optional)
Jira ticket says 
> When a user edits their displayName to be empty, it should get set to null

but decided that empty value is invalid



## Checklist

See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.





[DET-7518]: https://determinedai.atlassian.net/browse/DET-7518?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
